lib/components/map/itinerary-summary-overlay.tsx

```diff
@@ -0,0 +1,218 @@
+
+type ItinUniquePoint = {
+  itin: Itinerary
+  uniquePoint: Position
+}
+
+function getUniquePoint(
+  thisItin: ItinWithGeometry,
+  otherPoints: ItinUniquePoint[]
+): ItinUniquePoint {
+  const otherMidpoints = otherPoints.map((mp) => mp.uniquePoint)
+  let maxDistance = -Infinity
+  const line = thisItin.allLegGeometry
+  const centerOfLine = centroid(line).geometry.coordinates
+  let uniquePoint = centerOfLine
+
+  line.geometry.coordinates.forEach((point) => {
+    const totalDistance = otherMidpoints.reduce(
+      (prev, cur) => (prev += distance(point, cur)),
+      0
+    )
+
+    const selfDistance = distance(point, centerOfLine)
+    // maximize distance from all other points while minimizing distance to center of our own line
+    const averageDistance = totalDistance / otherMidpoints.length - selfDistance
+
+    if (averageDistance > maxDistance) {
+      maxDistance = averageDistance
+      uniquePoint = point
+    }
+  })
+  return { itin: thisItin, uniquePoint }
+}
+
+const ItinerarySummaryOverlay = ({
+  itins,
+  setActiveItinerary: setActive,
+  setVisibleItinerary: setVisible,
+  visible,
+  visibleItinerary
+}: Props) => {
+  // @ts-expect-error React context is populated dynamically
+  const { LegIcon } = useContext(ComponentContext)
+
+  let sharedTimeout: null | NodeJS.Timeout = null
+
+  if (!itins || !visible) return <></>
+  const mergedItins: ItinWithGeometry[] =
+    doMergeItineraries(itins).mergedItineraries.map(addItinLineString)
+  const midPoints = mergedItins.reduce<ItinUniquePoint[]>((prev, curItin) => {
+    prev.push(getUniquePoint(curItin, prev))
+    return prev
+  }, [])
+  // The first point is probably not well placed, so let's run the algorithm again
+  if (midPoints.length > 1) {
+    midPoints[0] = getUniquePoint(mergedItins[0], midPoints)
+  }
+
+  try {
+    return (
+      <>
+        {midPoints.map(
+          (mp, index) =>
+            // If no itinerary is hovered, show all of them. If one is selected, show only that one
+            // TODO: does this cause an issue with the merging? is the right itinerary selected?`
+            (visibleItinerary !== null && visibleItinerary !== undefined
+              ? visibleItinerary === index
+              : true) &&
+            mp.uniquePoint && (
+              <Marker
+                key={mp.itin.duration}
+                latitude={mp.uniquePoint[0]}
+                longitude={mp.uniquePoint[1]}
+                style={{ cursor: 'pointer' }}
+              >
+                <Card
+                  onClick={() => {
+                    setActive({ index })
+                  }}
+                  onMouseEnter={() => {
+                    sharedTimeout = setTimeout(() => {
+                      setVisible({ index })
+                    }, 150)
+                  }}
+                  onMouseLeave={() => {
+                    sharedTimeout && clearTimeout(sharedTimeout)
+                    setVisible({ index: null })
+                  }}
```

Comment:
Surround these handlers with `useCallback`

Reply:
I'm getting strange errors when I try this. But I think we should do this as part of a cleanup of this component